src/components/RichContenteditable/RichContenteditable.vue

```diff
@@ -0,0 +1,328 @@
+	</div>
+</template>
+<script>
+export default {
+	data() {
+		return {
+			message: '',
+			// You need to provide this for the inline
+			// mention to understand what to display or not.
+			userData: {
+				Test01: {
+					icon: 'icon-user',
+					id: 'Test01',
+					label: 'Test01',
+					source: 'users',
+					primary: true
+				},
+				Test02: {
+					icon: 'icon-user',
+					id: 'Test02',
+					label: 'Test02',
+					source: 'users'
+				}
+			}
+		}
+	},
+	methods: {
+		/**
+		* Do your own query to the autocompletion api.
+		* The returned data bellow is a fake data example.
+		* The callback expects the same format returned by the core/autocomplete/get ocs api endpoint!
+		* @param {string} search the query
+		* @param {Function} callback the callback to process the results with
+		*/
+		autoComplete(search, callback) {
+			return callback(Object.values(this.userData))
+		}
+	}
+}
+</script>
+```
+
+</docs>
+
+<template>
+	<div ref="contenteditable"
+		:class="{'rich-contenteditable__input--empty': isEmptyValue}"
+		:contenteditable="contenteditable"
+		:placeholder="placeholder"
+		class="rich-contenteditable__input"
+		@input="onInput"
+		@keydown.delete="onDelete"
+		@paste.stop.prevent="onPaste" />
+</template>
+
+<script>
+import Tribute from 'tributejs/dist/tribute.esm'
+import debounce from 'debounce'
+
+import { t } from '../../l10n.js'
+import AutoCompleteResult from './AutoCompleteResult'
+import richEditor from '../../mixins/richEditor/index'
+
+export default {
+	name: 'RichContenteditable',
+
+	mixins: [richEditor],
+
+	props: {
+		value: {
+			type: String,
+			required: true,
+		},
+		placeholder: {
+			type: String,
+			default: t('Write message, @ to mention someone …'),
+		},
+		autoComplete: {
+			type: Function,
+			required: true,
+		},
+		menuContainer: {
+			type: Element,
+			default: () => document.body,
+		},
+		/**
+		 * Is the content editable
+		 */
+		contenteditable: {
+			type: Boolean,
+			default: true,
+		},
+	},
+
+	data() {
+		return {
+			tribute: null,
+			options: {
+				fillAttr: 'id',
+				// Search against id and label (display name)
+				lookup: result => `${result.id} ${result.label}`,
+				// Where to inject the menu popup
+				menuContainer: this.menuContainer,
+				// Popup mention autocompletion templates
+				menuItemTemplate: item => this.renderComponentHtml(item.original, AutoCompleteResult),
+				// Hide if no results
+				noMatchTemplate: () => '<span class="hidden"></span>',
+				// Inner display of mentions
+				selectTemplate: item => this.genSelectTemplate(item.original?.id),
+				// Autocompletion results
+				values: this.debouncedAutoComplete,
+			},
+
+			// Represent the raw untrimmed text of the contenteditable
+			// serves no other purpose than to check whether the
+			// content is empty or not
+			localValue: '',
+		}
+	},
+
+	computed: {
+		isEmptyValue() {
+			return !this.localValue || this.localValue.trim() === ''
+		},
+
+		/**
+		 * Is this Firefox? 🙄
```

Comment:
computed prop in case it changes during runtime? :speak_no_evil:

Reply:
Not gonna lie, I gave it a thought before realising that was a joke :sob: :grin: